File: 0535-encode-and-decode-tinyurl/0535-encode-and-decode-tinyurl.py
class Codec:

    def __init__(self):
        self.decodeMap = {}
        self.base = "http://tinyurl.com/"
        self.current = 0

    def encode(self, longUrl: str) -> str:
        url = self.base + str(self.current)
        self.current += 1
        self.decodeMap[url] = longUrl
        return url

    def decode(self, shortUrl: str) -> str:
        return self.decodeMap[shortUrl]

# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(url))

File: 0535-encode-and-decode-tinyurl/0535-encode-and-decode-tinyurl.cpp
class Solution {
private:
    unordered_map<string, string> shortToLong;
    int nextShort = 0;
public:

    // Encodes a URL to a shortened URL.
    string encode(string longUrl) {
        string short_url = "http://tinyurl.com/" + to_string(nextShort++);
        shortToLong[short_url] = longUrl;
        return short_url;
    }

    // Decodes a shortened URL to its original URL.
    string decode(string shortUrl) {
        return shortToLong[shortUrl];
    }
};

// Your Solution object will be instantiated and called as such:
// Solution solution;
// solution.decode(solution.encode(url));